File: testdata/ruby/simple.rb
require 'json'
require_relative './helper'

module Greetings
  class Greeter
    def greet(name)
      "Hello, #{name}!"
    end

    private

    def format_name(name)
      name.capitalize
    end
  end

  class FormalGreeter < Greeter
    def greet(name)
      "Good day, #{format_name(name)}."
    end
  end
end

def standalone_function(input)
  greeter = Greetings::FormalGreeter.new
  greeter.greet(input)
end

File: testdata/lua/simple.lua
local json = require("json")

-- A simple greeter table (acts as a class)
local Greeter = {}
Greeter.__index = Greeter

function Greeter:new(name)
    local instance = setmetatable({}, Greeter)
    instance.name = name
    return instance
end

function Greeter:greet()
    return "Hello, " .. self.name .. "!"
end

-- A global function
function format_name(name)
    return string.upper(name)
end

-- A local helper function
local function helper(input)
    return input:lower()
end

-- Usage
local g = Greeter:new("World")
print(g:greet())
print(format_name("test"))
